experimental/oak_baremetal_channel/src/lib.rs

```diff
@@ -38,18 +44,20 @@ type FrameLength = u32;
 const FRAME_LENGTH_SIZE: usize = 4;
 static_assertions::assert_eq_size!([u8; FRAME_LENGTH_SIZE], FrameLength);
 
+// TODO(#2848): Move this onto the message layer, thereby conceptually
+// seperating the framing and messages layers.
 /// In request frames this field contains the id of the method that is invoked.
 /// In response frames this field contains the status code of the method invocation.
 type MethodOrStatus = u32;
 const METHOD_SIZE: usize = 4;
 static_assertions::assert_eq_size!([u8; METHOD_SIZE], MethodOrStatus);
 
-// TODO(#2848): Use this space to send a stream identifier that identifies
-// frames as belonging to the same message. Currently the frames of every message
-// must be sent as an uninterrupted seqeuence. A stream identifier would allow us
-// to interleave messages.
-/// Reserved space to maintain 8 byte alignement.
-const PADDING_SIZE: usize = 4;
+// TODO(#2848): Create a seperate invocation ID on the message layer to link
+// request and response messages, seperating the layers.
+/// Links the frames that constitute a message. Links request messages and
```

Comment:
Also, is multiplexing required for any use case already? If not, can we get away in the C++ implementation to assume no interleaving (i.e. always send all the frames for a given message before switching to the next one?) 

This PR (and anything else necessary for multiplexing) can still go ahead, as I think it will be useful, but I just wonder if in the short term we can get away without implementing all of this logic in C++ in order to get the stack up and running.

Reply:
Regarding the flakiness: since #2967 I am no longer experiencing that. I was unable to get it to stall even when running 100 000 iterations of the test. It is possible that the stalling was caused by some wrapping issue when doing flow control credit calculations with a stream buffer size set at the maximum value that the datatype could handle.

Reply:
> I think the terminology message id is a bit misleading here: this is really the id of an invocation, which is exemplified by the fact that (AFAICT) it is shared between the request and response message, so it does not uniquely identify a message (in my original IDL design I called it transaction id).

> Also see https://fuchsia.dev/fuchsia-src/reference/fidl/language/wire-format#transactional-messages for some more (non-normative) ideas.

> I think in our case we can stick with invocation instead of transaction for the time being, though we can keep thinking about it longer term (e.g. if we decide to support streaming multiple requests or responses instead of just unary invocations).

I don't disagree, the use of shared ID was based on the version of the design doc we reviewed together in the on demand slot. I agree that ideally it would be be better to use two separate IDs for linking, and for linkning requests/responses. 

Imo sensible names would be `MessageId` for the ID to link the frames that constitute a message, and `InvocationId`, for linking requests with responses. 

We can discuss it more and decide in a committed spec.